libs/sdk-js/src/react/stream.tsx

```diff
@@ -997,23 +1053,25 @@ export function useStream<
       // Assumption: we're setting the initial value
       // Used for instant feedback
       setStreamValues(() => {
-        const values = { ...historyValues };
+        const baseValues = options.initialValues ? { ...historyValues, ...options.initialValues } : historyValues;
 
```

Comment:
My take is that we should clear out the `initialValues` when we receive new `values` rather than always keeping those values around.